src/rlookup.c

```diff
@@ -125,7 +129,7 @@ size_t RLookup_GetLength(const RLookup *lookup, const RLookupRow *r, int *skipFi
     // we trust the shards to not send those fields.
     if (rule && ((rule->lang_field && strcmp(kk->name, rule->lang_field) == 0) ||
                   (rule->score_field && strcmp(kk->name, rule->score_field) == 0) ||
-                  (rule->score_field && strcmp(kk->name, rule->payload_field) == 0))) {
+                  (rule->payload_field && strcmp(kk->name, rule->payload_field) == 0))) {
```

Comment:
This is a bug fix no? We should track it and understand which versions are effected by it...

Reply:
@emmanuelkeller can you please review the comment about `payload_field`?

Reply:
@ashtul  This fix should go on master and on 2.0 (it it also exists there)